src/python/pants/backend/project_info/peek.py

```diff
@@ -237,6 +329,8 @@ async def get_target_data(
             for application in all_applicable_dep_rules
         }
 
+    # target_alias_to_goals_map = await _create_target_alias_to_goals_map() if subsys.include_goals else {}
```

Comment:
Errors out at runtime if I try to run this from here

Reply:
With what error?

Reply:
Essentially that this needs to be called in a goal rule, not in a regular rule

Reply:
```bash
 No installed rules return the type BuildTargetIdentifier, and it was not provided by potential callers of @rule(pants.bsp.util_rules.targets:212:resolve_bsp_build_target_identifier(BuildTargetIdentifier, BSPBuildTargets) -> BSPBuildTargetInternal).
    If that type should be computed by a rule, ensure that that rule is installed.
    If it should be provided by a caller, ensure that it is included in any relevant Query or Get.
  No installed rules return the type Specs, and it was not provided by potential callers of @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]).
    If that type should be computed by a rule, ensure that that rule is installed.
    If it should be provided by a caller, ensure that it is included in any relevant Query or Get.
  No installed rules return the type TargetRootsToFieldSetsRequest, and it was not provided by potential callers of @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]).
    If that type should be computed by a rule, ensure that that rule is installed.
    If it should be provided by a caller, ensure that it is included in any relevant Query or Get.
  No installed rules return the type _ParseOneBSPMappingRequest, and it was not provided by potential callers of @rule(pants.bsp.util_rules.targets:137:parse_one_bsp_mapping(_ParseOneBSPMappingRequest) -> BSPBuildTargetInternal).
    If that type should be computed by a rule, ensure that that rule is installed.
    If it should be provided by a caller, ensure that it is included in any relevant Query or Get.
  No source of dependency BSPBuildTargets for @rule(pants.bsp.util_rules.targets:212:resolve_bsp_build_target_identifier(BuildTargetIdentifier, BSPBuildTargets) -> BSPBuildTargetInternal). All potential sources were eliminated: []
  No source of dependency Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest]) for @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]). All potential sources were eliminated: []
  No source of dependency Get(FilteredTargets, [Specs]) for @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]). All potential sources were eliminated: []
  No source of dependency RegisteredTargetTypes for @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]). All potential sources were eliminated: []
  No source of dependency UnionMembership for @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]). All potential sources were eliminated: []
Traceback (most recent call last):
  File "/Users/sj/Developer/scratch/oss/pants-peek/src/python/pants/bin/daemon_pants_runner.py", line 133, in single_daemonized_run
    scheduler, options_initializer = self._core.prepare(options_bootstrapper, complete_env)
  File "/Users/sj/Developer/scratch/oss/pants-peek/src/python/pants/pantsd/pants_daemon_core.py", line 176, in prepare
    self._initialize(
  File "/Users/sj/Developer/scratch/oss/pants-peek/src/python/pants/pantsd/pants_daemon_core.py", line 119, in _initialize
    raise e
  File "/Users/sj/Developer/scratch/oss/pants-peek/src/python/pants/pantsd/pants_daemon_core.py", line 109, in _initialize
    self._scheduler = EngineInitializer.setup_graph(
  File "/Users/sj/Developer/scratch/oss/pants-peek/src/python/pants/init/engine_initializer.py", line 198, in setup_graph
    return EngineInitializer.setup_graph_extended(
  File "/Users/sj/Developer/scratch/oss/pants-peek/src/python/pants/init/engine_initializer.py", line 345, in setup_graph_extended
    scheduler = Scheduler(
  File "/Users/sj/Developer/scratch/oss/pants-peek/src/python/pants/engine/internals/scheduler.py", line 227, in __init__
    self._py_scheduler = native_engine.scheduler_create(
ValueError: Encountered 9 rule graph errors:
  No installed rules return the type BuildTargetIdentifier, and it was not provided by potential callers of @rule(pants.bsp.util_rules.targets:212:resolve_bsp_build_target_identifier(BuildTargetIdentifier, BSPBuildTargets) -> BSPBuildTargetInternal).
    If that type should be computed by a rule, ensure that that rule is installed.
    If it should be provided by a caller, ensure that it is included in any relevant Query or Get.
  No installed rules return the type Specs, and it was not provided by potential callers of @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]).
    If that type should be computed by a rule, ensure that that rule is installed.
    If it should be provided by a caller, ensure that it is included in any relevant Query or Get.
  No installed rules return the type TargetRootsToFieldSetsRequest, and it was not provided by potential callers of @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]).
    If that type should be computed by a rule, ensure that that rule is installed.
    If it should be provided by a caller, ensure that it is included in any relevant Query or Get.
  No installed rules return the type _ParseOneBSPMappingRequest, and it was not provided by potential callers of @rule(pants.bsp.util_rules.targets:137:parse_one_bsp_mapping(_ParseOneBSPMappingRequest) -> BSPBuildTargetInternal).
    If that type should be computed by a rule, ensure that that rule is installed.
    If it should be provided by a caller, ensure that it is included in any relevant Query or Get.
  No source of dependency BSPBuildTargets for @rule(pants.bsp.util_rules.targets:212:resolve_bsp_build_target_identifier(BuildTargetIdentifier, BSPBuildTargets) -> BSPBuildTargetInternal). All potential sources were eliminated: []
  No source of dependency Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest]) for @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]). All potential sources were eliminated: []
  No source of dependency Get(FilteredTargets, [Specs]) for @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]). All potential sources were eliminated: []
  No source of dependency RegisteredTargetTypes for @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]). All potential sources were eliminated: []
  No source of dependency UnionMembership for @rule(pants.engine.internals.specs_rules:484:find_valid_field_sets_for_target_roots(TargetRootsToFieldSetsRequest, Specs, UnionMembership, RegisteredTargetTypes) -> TargetRootsToFieldSets, gets=[Get(FilteredTargets, [Specs]), Get(FieldSetsPerTarget, [FieldSetsPerTargetRequest])]). All potential sources were eliminated: []
```